src/lib/bootloader/grub2pwd.rb

```diff
@@ -1,63 +1,146 @@
 require "yast"
 
-# class is responsible for detection, encryption and writing of grub2 password protection
-class GRUB2Pwd
-  YAST_BASH_PATH = Yast::Path.new(".target.bash_output")
-  PWD_ENCRYPTION_FILE = "/etc/grub.d/42_password"
+Yast.import "Stage"
 
-  def used?
-    Yast.import "FileUtils"
+module Bootloader
+  # class is responsible for detection, encryption and writing of grub2 password protection
+  class GRUB2Pwd
+    # @!attribute used
+    #   @return [bool] specifies if password protection enabled
+    #
+    # @!attribute unrestricted
+    #   @return [bool] specifies if unrestricted password protection should be used
```

Comment:
even Boolean is not known class in ruby, there is only TrueClass and FalseClass.
Well, unrestricted is official grub2 term. I expect that this code will be modified only by someone who knows grub2 terminology.